yamux/src/connection/stream.rs

```diff
@@ -351,6 +360,12 @@ impl AsyncWrite for Stream {
         let mut frame = Frame::data(self.id, body).expect("body <= u32::MAX").left();
         self.add_flag(frame.header_mut());
         log::trace!("{}/{}: write {} bytes", self.conn, self.id, n);
+
+        // technically, the frame hasn't been sent yet on the wire but from the perspective of this data structure, we've queued the frame for sending
+        if frame.header().flags().contains(ACK) {
+            self.set_acknowledged();
+        }
```

Comment:
This is confusing to me. Do I understand correctly that this sets an **inbound** stream to be acknowledged? Why would the local state machine care whether an inbound stream is acknowledged or not?

Reply:
We can omit this but it means that the flag is only used by outbound streams which IMO is even more confusing.

I guess I could rename it? Or store it alongside with it being client/server.

Reply:
Ah, I remember why I am doing it for both. I needed that for the test around timing, i.e. when a stream gets acknowledged.

Reply:
That is fine by me. Though could you please document it for future us?